src/marqo/tensor_search/add_docs.py

```diff
@@ -88,10 +88,10 @@ def threaded_download_and_preprocess_content(allocated_docs: List[dict],
                 if field not in tensor_fields:
                     continue
                 if isinstance(doc[field], str) or force_download:
-                    modality = infer_modality(doc[field])
-                    if (modality == Modality.IMAGE and is_no_index) or (
-                        modality == Modality.IMAGE and is_unstructured_index) or (
-                        is_structured_index and media_field_types_mapping[field] == FieldType.ImagePointer):
+                    inferred_modality = infer_modality(doc[field])
```

Comment:
Why don't we catch the requests error here and return the error message to the user?

Reply:
it might lead to an internal error which would fail the whole batch

catching it and passing to all the places which call infer_modality would take a bit of time. 

I think add_docs logic right now catches this failure but only returns something generic like 

`f"Error processing media file {doc}, detected as text or failed to determine modality"`

Reply:
There's surely a better way in the future though

Reply:
scratch the above. catching the error